From 13354b602f23c9c4ec699dddeb4b8228beaad011 Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Wed, 13 Jan 2016 11:16:01 +0200
Subject: [PATCH 0683/1240] doc: board: updated board setup doc for
 Armada-7040-RZ

- Changed the folder and file name
- Added template section for board setup
- Added name for each setup

Change-Id: I8e939ad3140c0596ee766f0286bc9d0381696ed2
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/26754
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 doc/mvebu/a8k/board_options.txt             | 32 ----------
 doc/mvebu/boards/armada7040-rz-db-setup.txt | 99 +++++++++++++++++++++++++++++
 2 files changed, 99 insertions(+), 32 deletions(-)
 delete mode 100644 doc/mvebu/a8k/board_options.txt
 create mode 100644 doc/mvebu/boards/armada7040-rz-db-setup.txt

diff --git a/doc/mvebu/a8k/board_options.txt b/doc/mvebu/a8k/board_options.txt
deleted file mode 100644
index 700d787..0000000
--- a/doc/mvebu/a8k/board_options.txt
+++ /dev/null
@@ -1,32 +0,0 @@
-Board options:
---------------
-
-The below describes the board options for Armada-8k/7k
-For each board, the below defines the list of configurations that U-Boot support, using device tree blob.
-
-Option list of Armada-8040-RZ:
-
-option #	|	1	|	2	|	3	|	4	|	5	|	6	|	7	|
----------------------------------------------------------------------------------------------------------------------------------
-SLM #		| SML1524	| SLM1521	| SLM1521	| SLM1521	| SLM1521	| SLM1522	| SLM1522	|
----------------------------------------------------------------------------------------------------------------------------------
-SerDes:	lane0	| sgmii2	| pex_x4_l0	| pex_x4_l0	| pex_x4_l0	| pex_x4_l0	| sata1		| pex_x4_l0	|
-	lane1	| usb3h0	| pex_x4_l1	| sata0		| usb3h0	| usb3h0	| sata0		| sata0		|
-	lane2	| sgmii0	| pex_x4_l2	| kr		| kr		| sata0		| sgmii0	| sgmii0	|
-	lane3	| sata1		| pex_x4_l3	| sata1		| sata1		| sata1		| sgmii2	| sgmii2	|
-	lane4	| usb3h1	| pex_x1_0	| pex_x1_0	| usb3h1	| usb3h1	| rxauil0	| rxauil0	|
-	lane5	| pex_x1_1	| pex_x1_1	| pex_x1_1	| pex_x1_1	| pex_x1_1	| rxauil1	| rxauil1	|
----------------------------------------------------------------------------------------------------------------------------------
-MPPs: 	CP-SPI0	| N/C		| N/C		| N/C		| N/C		| N/C		| N/C		| N/C		|
-	CP-SPI1	| [13-16]	| [13-16]	| [13-16]	| N/C		| [13-16]	| [13-16]	| [13-16]	|
-	NAND	| N/C		| N/C		| N/C		| [13,17-27]	| N/C		| N/C		| N/C		|
-	RGMII0	| N/C		| [0-12]	| [0-12]	| [0-12]	| [0-12]	| N/C		| N/C		|
-	RGMII1	| N/C		| [44-55]	| [44-55]	| [44-55]	| [44-55]	| N/C		| N/C		|
-	CP-UART0| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	|
-	CP-UART1| [40-41]	| [40-41]	| [40-41]	| [40-41]	| [40-41]	| [40-41]	| [40-41]	|
-	SD	| [56-62]	| [56-62]	| [56-62]	| [56-62]	| [56-62]	| [56-62]	| [56-62]	|
-	TDM	| [0-11]	| N/C		| N/C		| N/C		| N/C		| N/C		| N/C		|
-	TWSI	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	|
-	SATA0	| [36]		| N/C		| [36]		| N/C		| [36]		| [36]		| [36]		|
-	SATA1	| [28]		| N/C		| [28]		| [28]		| [28]		| [28]		| N/C		|
----------------------------------------------------------------------------------------------------------------------------------
diff --git a/doc/mvebu/boards/armada7040-rz-db-setup.txt b/doc/mvebu/boards/armada7040-rz-db-setup.txt
new file mode 100644
index 0000000..f4e1441
--- /dev/null
+++ b/doc/mvebu/boards/armada7040-rz-db-setup.txt
@@ -0,0 +1,99 @@
+Armada7040-RZ development board setup:
+--------------------------------------
+
+The Armada7040-RZ Development Board (DB) is a configurable board designed to test several
+combinations of SOC configurations. U-BOOT supports a defined set of those configurations
+via different device tree files.
+
+This document describes the board modifications required to setup each configuration and the
+interfaces supported (or disabled in each one).
+
+To switch to a different configuration, follow the instructions in the "board setup" section.
+Once completed change the FDT file used by U-BOOT by running the U-BOOT command "fdt_config".
+
+
+Board Setup
+------------
+
+## setup 1: Router ###
+
+  Serdes site
+   - module: SLM1524
+   - Module DIP switch
+
+
+## setup 2: All PCI ###
+
+  Serdes site
+   - module: SLM1521
+   - Module DIP switch
+
+
+## setup 3: KR ###
+
+  Serdes site
+   - module: SLM1521
+   - Module DIP switch
+
+## setup 4: NAND ###
+
+  Serdes site
+   - module: SLM1521
+   - Module DIP switch
+
+## setup 5: RGMII only ###
+
+  Serdes site
+   - module: SLM1521
+   - Module DIP switch
+
+## setup 6: RXAUI ###
+
+  Serdes site
+   - module: SLM1522
+   - Module DIP switch
+
+## setup 7: RXAUI PCI  ###
+
+  Serdes site
+   - module: SLM1522
+   - Module DIP switch
+
+
+The tables below summarizes the interface configuration of each setup
+
+Serdes PHY configuration
+------------------------
+
+setup  #	|	1	|	2	|	3	|	4	|	5	|	6	|	7	|
+name   #	|     router 	|    all pci	|	KR	|     NAND	|   rmgii only	|    rxaui	|  rxaui-pci	|
+---------------------------------------------------------------------------------------------------------------------------------
+SerDes:	lane0	| sgmii2	| pex_x4_l0	| pex_x4_l0	| pex_x4_l0	| pex_x4_l0	| sata1		| pex_x4_l0	|
+	lane1	| usb3h0	| pex_x4_l1	| sata0		| usb3h0	| usb3h0	| sata0		| sata0		|
+	lane2	| sgmii0	| pex_x4_l2	| kr		| kr		| sata0		| sgmii0	| sgmii0	|
+	lane3	| sata1		| pex_x4_l3	| sata1		| sata1		| sata1		| sgmii2	| sgmii2	|
+	lane4	| usb3h1	| pex_x1_0	| pex_x1_0	| usb3h1	| usb3h1	| rxauil0	| rxauil0	|
+	lane5	| pex_x1_1	| pex_x1_1	| pex_x1_1	| pex_x1_1	| pex_x1_1	| rxauil1	| rxauil1	|
+---------------------------------------------------------------------------------------------------------------------------------
+
+
+Multi purpose pin configurations
+--------------------------------
+
+setup  #	|	1	|	2	|	3	|	4	|	5	|	6	|	7	|
+name   #	|     router 	|    all pci	|	KR	|     NAND	|   rmgii only	|    rxaui	|  rxaui-pci	|
+---------------------------------------------------------------------------------------------------------------------------------
+
+MPPs: 	CP-SPI0	| N/C		| N/C		| N/C		| N/C		| N/C		| N/C		| N/C		|
+	CP-SPI1	| [13-16]	| [13-16]	| [13-16]	| N/C		| [13-16]	| [13-16]	| [13-16]	|
+	NAND	| N/C		| N/C		| N/C		| [13,17-27]	| N/C		| N/C		| N/C		|
+	RGMII0	| N/C		| [0-12]	| [0-12]	| [0-12]	| [0-12]	| N/C		| N/C		|
+	RGMII1	| N/C		| [44-55]	| [44-55]	| [44-55]	| [44-55]	| N/C		| N/C		|
+	CP-UART0| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	|
+	CP-UART1| [40-41]	| [40-41]	| [40-41]	| [40-41]	| [40-41]	| [40-41]	| [40-41]	|
+	SD	| [56-62]	| [56-62]	| [56-62]	| [56-62]	| [56-62]	| [56-62]	| [56-62]	|
+	TDM	| [0-11]	| N/C		| N/C		| N/C		| N/C		| N/C		| N/C		|
+	TWSI	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	|
+	SATA0	| [36]		| N/C		| [36]		| N/C		| [36]		| [36]		| [36]		|
+	SATA1	| [28]		| N/C		| [28]		| [28]		| [28]		| [28]		| N/C		|
+---------------------------------------------------------------------------------------------------------------------------------
-- 
1.9.1

